src/json_schema/parsing.rs

```diff
@@ -329,6 +340,21 @@ impl<'a> Parser<'a> {
             _ => Err(JsonSchemaParserError::UnsupportedType(Box::from(
                 instance_type,
             ))),
+        };
+
+        if instance_types.len() == 1 {
+            parse_instance_type(instance_types[0])
+        } else {
+            let xor_patterns = instance_types
+                .into_iter()
+                .map(|instance_type| {
+                    let sub_regex = parse_instance_type(instance_type)?;
+
+                    Ok(format!(r"(?:{})", sub_regex))
+                })
+                .collect::<Result<Vec<String>>>()?;
+
+            Ok(format!(r"({})", xor_patterns.join("|")))
```

Comment:
Hi @bnjjj 👋 

Thank you for working on this!

This code needs the following improvements for readability and efficiency (it unnecessarily creates new vectors):
1) add `parse_type_string` (with old `parse_type` logic ) and `parse_type_array` (new xor business) methods
2) update `parse_type` with match on type:
```
    match obj.get("type") {
        Some(Value::String(instance_type)) => self.parse_type_string(instance_type, obj),
        Some(Value::Array(instance_types)) => self.parse_type_array(instance_types, obj),
        _ => Err(JsonSchemaParserError::TypeMustBeAString),
    }
```